flexmeasures/data/models/planning/storage.py

```diff
@@ -470,12 +462,17 @@ def deserialize_flex_config(self):
                 self.flex_model["soc-unit"] = self.sensor.unit
             elif self.sensor.unit in ("MW", "kW"):
                 self.flex_model["soc-unit"] = self.sensor.unit + "h"
+            else:
+                # todo: raise? Surely we must have a unit.
+                pass
 
         self.ensure_soc_min_max()
 
-        # Now it's time to check if our flex configurations holds up to schemas
+        # Now it's time to check if our flex configuration holds up to schemas
         self.flex_model = StorageFlexModelSchema(
-            start=self.start, sensor=self.sensor
+            start=self.start,
+            sensor=self.sensor,
+            default_soc_unit=self.flex_model["soc-unit"],
```

Comment:
Not trivially. The fields need the soc-unit upon initialization, whereas we're passing (the rest of) the flex-model upon loading. But I did move some of the soc-unit logic into the schema, so that should look a bit cleaner already.